src/mngt/ocf_mngt_flush.c

```diff
@@ -974,6 +974,9 @@ int ocf_mngt_cache_cleaning_set_policy(ocf_cache_t cache, ocf_cleaning_t type)
 
 	ocf_metadata_unlock(cache, OCF_METADATA_WR);
 
+	if (type != ocf_cleaning_nop)
+		ocf_kick_cleaner(cache);
```

Comment:
I would move this to cleaning policy initialize() op. It's cleaning policy responsibility to decide whether it wants cleaner to be kicked or not.

Reply:
Moving it there would cause a problem because in cache attach procedure initialize() is called before cleaner thread is started.

Reply:
@mmichal10 
What about starting cleaner earlier? We can define the rule, that ocf_cleaner_run() shall not be called before first kick. Besides, even if it was, the thread would immediately go to sleep because the cache is locked, so nothing wrong could happen.